src/params/types.ts

```diff
@@ -41,6 +41,33 @@ export abstract class Expression<T extends string | number | boolean | string[]>
   }
 }
 
+/**
+ * A CEL expression which represents an internal Firebase variable. This class
+ * cannot be instantiated by developers, but we provide several canned instances
+ * of it to make available params that will never have to be defined at
+ * deployment time, and can always be read from process.env.
+ */
+export class InternalExpression extends Expression<string> {
```

Comment:
Should this be marked @hidden or @beta or @alpha or whatever so we don't expose it to our end user?

I wish we can use @internal, but I'm afraid that might break user code that tries to reference variables with that type 🤔

Reply:
No, it should be @internal and the variable should be declared as an Expression<string>. Type erasure is great for hiding implementation details and not tying our hands to them.

Reply:
That's a good point. Done.